File: id/src/main/java/com/simprints/id/activities/matching/MatchingContract.java
package com.simprints.id.activities.matching;


import android.content.Intent;

import com.simprints.id.activities.BasePresenter;
import com.simprints.id.activities.BaseView;
import com.simprints.id.model.ALERT_TYPE;

interface MatchingContract {

    interface View extends BaseView<Presenter> {

        void setIdentificationProgress(int progress);

        void setVerificationProgress();

        void setIdentificationProgressLoadingStart();

        void setIdentificationProgressMatchingStart(int matchSize);

        void setIdentificationProgressReturningStart();

        void setIdentificationProgressFinished(int returnSize, int tier1Or2Matches, int tier3Matches, int tier4Matches, int matchingEndWaitTimeMillis);

        void launchAlert(ALERT_TYPE alertType);

        void makeToastMatchNotRunning(String text);

        void doSetResult(int resultCode, Intent resultData);

        void doFinish();
    }

    interface Presenter extends BasePresenter {

    }
}

File: fingerprint/src/main/java/com/simprints/fingerprint/activities/matching/MatchingContract.kt
package com.simprints.fingerprint.activities.matching


import android.content.Intent
import com.simprints.fingerprint.activities.BasePresenter
import com.simprints.fingerprint.activities.BaseView
import com.simprints.fingerprint.orchestrator.domain.ResultCode


interface MatchingContract {

    interface View : BaseView<Presenter> {

        fun setIdentificationProgress(progress: Int)

        fun setVerificationProgress()

        fun setIdentificationProgressLoadingStart()

        fun setIdentificationProgressMatchingStart(matchSize: Int)

        fun setIdentificationProgressReturningStart()

        fun setIdentificationProgressFinished(returnSize: Int, tier1Or2Matches: Int, tier3Matches: Int, tier4Matches: Int, matchingEndWaitTimeMillis: Int)

        fun launchAlertActivity()

        fun makeToastMatchFailed()

        fun setResultAndFinish(resultCode: ResultCode, resultData: Intent?)

        fun doSetResult(resultCode: ResultCode, resultData: Intent?)

        fun doFinish()
    }

    interface Presenter : BasePresenter {

        fun handleBackPressed()

        fun dispose()
    }
}
